### PR TITLE
Ban importing `Match` and `Pattern` from `typing` (import from `re` instead)

### LINKED ISSUE #46: Suggest using collections.abc etc, not typing (deprecated since 3.9)
Affects:

* [x] `AbstractSet` (as `collections.abc.Set`)
* [X] `AsyncContextManager` (as `contextlib.AbstractAsyncContextManager`)
* [x] `AsyncGenerator`
* [x] `AsyncIterable`
* [x] `AsyncIterator`
* [x] `Awaitable`
* [x] `ByteString`
* [x] `Callable`
* [X] `ChainMap` (as `collections.ChainMap`)
* [x] `Collection`
* [x] `Container`
* [x] `ContextManager` (as `contextlib.AbstractContextManager`)
* [x] `Coroutine`
* [X] `Counter` (as `collections.Counter`)
* [X] `DefaultDict` (as `collections.defaultdict`)
* [X] `Deque` (as `collections.Deque`)
* [x] `Generator`
* [x] `Hashable`
* [x] `ItemsView`
* [x] `Iterable`
* [x] `Iterator`
* [x] `KeysView`
* [x] `Mapping`
* [x] `MappingView`
* [ ] `Match` (as `re.Match`)
* [x] `MutableMapping`
* [x] `MutableSequence`
* [x] `MutableSet`
* [x] `OrderedDict` (as `collections.OrderedDict`)
* [ ] `Pattern` (as `re.Pattern`)
* [x] `Reversible`
* [x] `Sequence`
* [x] `Sized`
* [x] `ValuesView`

### PR DESCRIPTION
**Pull request by AlexWaygood**
Closes #46

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ currently emitted:
 | Y019 | Certain kinds of methods should use `_typeshed.Self` instead of defining custom `TypeVar`s for their return annotation. This check currently applies for instance methods that return `self`, class methods that return an instance of `cls`, and `__new__` methods.
 | Y020 | Quoted annotations should never be used in stubs.
 | Y021 | Docstrings should not be included in stubs.
-| Y022 | Imports linting: use typing-module aliases to stdlib objects as little as possible (e.g. `builtins.list` over `typing.List`, `collections.Counter` over `typing.Counter`, etc.).
+| Y022 | Use typing-module aliases to stdlib objects as little as possible (e.g. `builtins.list` over `typing.List`, `collections.OrderedDict` over `typing_extensions.OrderedDict`, etc.).
```
Comment (srittau) [inline README.md:58]: Why use `typing_extensions` here? I think `typing` is more "canonical", easier to understand, and more succinct.

**Review by srittau — APPROVED** [review]
LGTM, one suggestion though.

File: README.md (hunk 1 above)
Comment (AlexWaygood) [reply README.md:58]: I realised we didn't mention anywhere that Y022 bans various imports from `typing_extensions` as well as various imports from `typing`, and that felt like something which might be useful to make clear. But I can reword this if the current phrasing is just confusing!

Comment (AlexWaygood) [reply README.md]: How's it look after 1bf878b?

**Pull request merged**